src/dialogs/more-info/components/climate/ha-more-info-climate-temperature.ts

```diff
@@ -0,0 +1,534 @@
+import { classMap } from "lit/directives/class-map";
+import { styleMap } from "lit/directives/style-map";
+import { computeAttributeValueDisplay } from "../../../../common/entity/compute_attribute_display";
+import { stateActive } from "../../../../common/entity/state_active";
+import { stateColorCss } from "../../../../common/entity/state_color";
+import { supportsFeature } from "../../../../common/entity/supports-feature";
+import { clamp } from "../../../../common/number/clamp";
+import { formatNumber } from "../../../../common/number/format_number";
+import { debounce } from "../../../../common/util/debounce";
+import "../../../../components/ha-control-circular-slider";
+import "../../../../components/ha-outlined-icon-button";
+import "../../../../components/ha-svg-icon";
+import {
+  CLIMATE_HVAC_ACTION_TO_MODE,
+  ClimateEntity,
+  ClimateEntityFeature,
+} from "../../../../data/climate";
+import { UNAVAILABLE } from "../../../../data/entity";
+import { HomeAssistant } from "../../../../types";
+
+type Target = "value" | "low" | "high";
+
+@customElement("ha-more-info-climate-temperature")
+export class HaMoreInfoClimateTemperature extends LitElement {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @property({ attribute: false }) public stateObj!: ClimateEntity;
+
+  @state() private _targetTemperature: Partial<Record<Target, number>> = {};
+
+  @state() private _selectTargetTemperature: Target = "low";
+
+  protected willUpdate(changedProp: PropertyValues): void {
+    super.willUpdate(changedProp);
+    if (changedProp.has("stateObj")) {
+      this._targetTemperature = {
+        value: this.stateObj.attributes.temperature,
+        low: this.stateObj.attributes.target_temp_low,
+        high: this.stateObj.attributes.target_temp_high,
+      };
+    }
+  }
+
+  private get _step() {
+    return (
+      this.stateObj.attributes.target_temp_step ||
+      (this.hass.config.unit_system.temperature.indexOf("F") === -1 ? 0.5 : 1)
+    );
+  }
+
+  private get _min() {
+    return this.stateObj.attributes.min_temp;
+  }
+
+  private get _max() {
+    return this.stateObj.attributes.max_temp;
+  }
+
+  private _valueChanged(ev: CustomEvent) {
+    const value = (ev.detail as any).value;
+    if (isNaN(value)) return;
+    const target = ev.type.replace("-changed", "");
+    this._targetTemperature = {
+      ...this._targetTemperature,
+      [target]: value,
+    };
+    this._selectTargetTemperature = target as Target;
+    this._callService(target);
+  }
+
+  private _valueChanging(ev: CustomEvent) {
+    const value = (ev.detail as any).value;
+    if (isNaN(value)) return;
+    const target = ev.type.replace("-changing", "");
+    this._targetTemperature = {
+      ...this._targetTemperature,
+      [target]: value,
+    };
+    this._selectTargetTemperature = target as Target;
+  }
+
+  private _debouncedCallService = debounce(
+    (target: Target) => this._callService(target),
+    2000
```

Comment:
2 seconds seems long, wouldn't 1 sec be enough?

Reply:
I reused the duration from the old control but we can reduce it 🙂